ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -956,8 +959,23 @@ boolean getIndexEntriesSince(StoreKey key, FindEntriesCondition findEntriesCondi
    * @param entries the entries to eliminate duplicates from.
    */
   private void eliminateDuplicates(List<IndexEntry> entries) {
-    Set<StoreKey> setToFindDuplicate = new HashSet<StoreKey>();
-    ListIterator<IndexEntry> iterator = entries.listIterator(entries.size());
+    Set<StoreKey> setToFindDuplicate = new HashSet<>();
```

Comment:
As `IndexEntry`s for same key are built from one 'ConcurrentSkipListSet<IndexValue>' now, looks like we can do `eliminateDuplicates()` in line 936~938,  to save one extra iteration for duplicates removal.

Reply:
But elimination within each set will have to follow the two iterations right ? But will consider how this can be done differently to avoid the loops

Reply:
Assuming there can only be one put, and one delete entry for a key in the list, I think you could simplify this method as follows (though this still does the two iterations):
```java
{
  Set<StoreKey> setToFindDuplicate = new HashSet<>();
  entries.removeIf(e -> !e.getValue().isFlagSet(IndexValue.Flags.Delete_Index) && 
                        !setToFindDuplicate.add(e.getKey())); //removes if already seen.
  setToFindDuplicate.clear();
  entries.removeIf(e -> setToFindDuplicate.add(e.getKey())); //removes if not seen.
}
```

Reply:
Could improve the first iteration but couldn't find a good substitute (that reduced LOC or didn't create new objects) for the reverse iteration